### DISCUSSION TITLE
Sbt by default runs suites in parallel, using its own thread pool.

### DISCUSSION THREAD

**Discussion by arbitrary-dev** [sbt/sbt#6313] [Q&A] [answered]
Got this from Scalatest docs.

Currently have issues in a big project, whenever I run all test suites, some of them hang and as result fail.

```
fork in Test := false
```
…so tests are run in parallel, I suppose.

Maybe you can recommend some JVM or SBT setting to tune in this case?

**Comment by eed3si9n** [comment, marked as answer]
> Maybe you can recommend some JVM or SBT setting to tune in this case?

This is too broad to answer. If you have purely computational unit tests and property-based tests then you can go parallel all the way. Otherwise you might need to fork. It's really case by case and how much memory you're giving to the build.

- https://www.scala-sbt.org/1.x/docs/Parallel-Execution.html
- https://www.scala-sbt.org/1.x/docs/Testing.html
- https://www.scala-sbt.org/1.x/docs/In-Process-Classloaders.html
